### ISSUE TITLE
ETAPA 1

### ISSUE THREAD

**Issue by willianemanoel** [willianemanoel/AI-MULTI---GTI-#1]
import pandas as pd
import numpy as np

# Definindo as variáveis
np.random.seed(0)
dates = pd.date_range(start='2023-01-01', end='2023-12-31', periods=1000)
vendas = np.random.normal(loc=1000, scale=300, size=1000).round(2)
produtos = np.random.randint(1, 10, size=1000)
tipos = np.random.choice(['Eletrônicos', 'Roupas', 'Livros', 'Alimentos'], size=1000)
canais = np.random.choice(['Online', 'Loja física'], size=1000)

# Criando o DataFrame
df = pd.DataFrame({'Data': dates, 'Venda': vendas, 'Produtos': produtos, 'Tipo': tipos, 'Canal': canais})

# Salvando o DataFrame em um arquivo CSV
df.to_csv('dataset_vendas.csv', index=False)

# Importar bibliotecas necessárias
import pandas as pd

# Carregar o dataset
df = pd.read_csv('dataset_vendas.csv')

# Verificar as primeiras linhas do dataset
print("Primeiras linhas do dataset:")
print(df.head())

# Verificar valores ausentes
print("\nValores ausentes no dataset:")
print(df.isnull().sum())

# Verificar valores duplicados
print("\nNúmero de valores duplicados no dataset:", df.duplicated().sum())

# Verificar outliers nos valores de venda
import seaborn as sns
import matplotlib.pyplot as plt

plt.figure(figsize=(10, 6))
sns.boxplot(x='Venda', data=df)
plt.title('Boxplot dos Valores de Venda')
plt.show()

# Calcular estatísticas básicas
print("\nEstatísticas básicas do dataset:")
print(df.describe())

# Calcular a mediana para cada tipo de produto
print("\nMediana das vendas por tipo de produto:")
print(df.groupby('Tipo')['Venda'].median())

# Calcular o desvio padrão para cada canal de vendas
print("\nDesvio padrão das vendas por canal de vendas:")
print(df.groupby('Canal')['Venda'].std())

# Importar bibliotecas necessárias
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error

# Escolher variáveis independentes (X) e variável dependente (y)
X = df[['Produtos']]  # Variável independente (número de produtos vendidos)
y = df['Venda']  # Variável dependente (valor da venda)

# Dividir os dados em conjuntos de treinamento e teste
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Instanciar o modelo de regressão linear
model = LinearRegression()

# Treinar o modelo
model.fit(X_train, y_train)

# Fazer previsões
y_pred = model.predict(X_test)

# Avaliar o desempenho do modelo
mse = mean_squared_error(y_test, y_pred)
print("\nErro quadrático médio (MSE):", mse)
print("Coeficiente de determinação (R²):", model.score(X_test, y_test))

# Importar bibliotecas necessárias
import matplotlib.pyplot as plt

# Gráfico de dispersão: número de produtos vendidos vs. valor da venda
plt.figure(figsize=(10, 6))
plt.scatter(df['Produtos'], df['Venda'], alpha=0.5)
plt.title('Relação entre Número de Produtos Vendidos e Valor da Venda')
plt.xlabel('Número de Produtos Vendidos')
plt.ylabel('Valor da Venda')
plt.grid(True)
plt.show()

# Gráfico de barras: distribuição dos tipos de produtos vendidos
plt.figure(figsize=(10, 6))
df['Tipo'].value_counts().plot(kind='bar')
plt.title('Distribuição dos Tipos de Produtos Vendidos')
plt.xlabel('Tipo de Produto')
plt.ylabel('Quantidade')
plt.grid(axis='y')
plt.show()

# Gráfico de linha: tendência das vendas ao longo do tempo
plt.figure(figsize=(10, 6))
df.groupby('Data')['Venda'].sum().plot()
plt.title('Tendência das Vendas ao Longo do Tempo')
plt.xlabel('Data')
plt.ylabel('Valor Total de Vendas')
plt.grid(True)
plt.show()

# Gráfico de caixa (boxplot): distribuição dos valores de venda para cada tipo de produto
plt.figure(figsize=(10, 6))
sns.boxplot(x='Tipo', y='Venda', data=df)
plt.title('Distribuição dos Valores de Venda por Tipo de Produto')
plt.xlabel('Tipo de Produto')
plt.ylabel('Valor da Venda')
plt.grid(True)
plt.show()

# Gráfico de barras: distribuição dos canais de vendas
plt.figure(figsize=(8, 5))
df['Canal'].value_counts().plot(kind='bar', color='skyblue')
plt.title('Distribuição dos Canais de Vendas')
plt.xlabel('Canal de Vendas')
plt.ylabel('Número de Vendas')
plt.grid(axis='y')
plt.show()

# Histograma: distribuição dos valores de venda
plt.figure(figsize=(10, 6))
plt.hist(df['Venda'], bins=20, color='salmon', edgecolor='black')
plt.title('Distribuição dos Valores de Venda')
plt.xlabel('Valor da Venda')
plt.ylabel('Frequência')
plt.grid(True)
plt.show()
